### ISSUE TITLE
FileUpload: Does not work properly

### ISSUE THREAD

**Issue by ivanpajon** [primefaces/primereact#5648]
### Describe the bug

I am working on a project built with Next 14, and using
```tsx
<form action={formAction}>
    <FileUpload
        id="fileUpload"
        name="fileUpload"
        mode="advanced"
        accept="image/*"
        maxFileSize={1000000}
    />

    <Button label="Submit" />
</form>
```
in which **formAction** is a **Server Action**, when submit the form it never send the file. Server Action receives a FormData as parameter, and in this case never has file attached.

I realized that **id** and **name** attributes are not passed properly to the internal hidden input that uses FileUpload component.
![image](https://github.com/primefaces/primereact/assets/17519714/41a12a00-17e9-41e9-8669-eb7494918755)

I tried some workarounds to get it work with no result.
1. First, set name using _FileUploadPassThroughOptions_, it set attribute properly but when submit form it sends something strange that not matches with the selected file.
```tsx
{/* This FileUpload send wrong file in fixedName instead of fileUploadFixedName */}
<FileUpload
    id="fileUploadFixedName"
    name="fileUploadFixedName"
    mode="advanced"
    accept="image/*"
    maxFileSize={1000000}
    pt={{ input: { name: 'fixedName' } }}
/>
```
![image](https://github.com/primefaces/primereact/assets/17519714/c722d87f-9b53-4812-9883-19ecdf343b58)
![image](https://github.com/primefaces/primereact/assets/17519714/c4973895-3d0c-4840-85ea-86e43a95057b)
![image](https://github.com/primefaces/primereact/assets/17519714/e9501a1a-452c-4cbf-9082-4f6d5a20049d)

2. I can set a 'dynamic' name to the input in the following way but with same results.
```tsx
const dynamicName = 'someDynamicName'; // Dynamic input name
{/* This FileUpload send wrong file with key value of dynamicName constant instead of fileUploadDynamicName */}
<FileUpload
    id="fileUploadDynamicName"
    name="fileUploadDynamicName"
    mode="advanced"
    accept="image/*"
    maxFileSize={1000000}
    pt={{ input: { name: dynamicName } }}
/>
```
![image](https://github.com/primefaces/primereact/assets/17519714/6442328f-0e04-490a-9a2c-4579c4094eda)
![image](https://github.com/primefaces/primereact/assets/17519714/8f4a3a84-969d-406b-b734-ead61f94b771)

3. Finally! I got a working solution by passing a template to the FileUpload's _onSelect_ property. In the template I receive the selected files and set them to a hidden input.
```tsx
const inputRef = useRef<HTMLInputElement>(null);
const onTemplateSelect = (event: FileUploadSelectEvent) => {
    const dataTransfer = new DataTransfer();
    for (const file of event.files) {
        dataTransfer.items.add(file);
    }
    inputRef.current!.files = dataTransfer.files;
};
{/* This input send correct file in correctFile setted by wrongFile FileUpload */}
<input ref={inputRef} name="correctFile" type="file" className="hidden" />
{/* This FileUpload send wrong file in wrongFile, but set the correct file in correctFile */}
<FileUpload
    mode="advanced"
    accept="image/*"
    maxFileSize={1000000}
    onSelect={onTemplateSelect}
    pt={{ input: { name: 'wrongFile' } }}
/>
```
![image](https://github.com/primefaces/primereact/assets/17519714/75f953b0-0e8d-448c-af0b-bb32dddd6d25)
![image](https://github.com/primefaces/primereact/assets/17519714/66b405aa-6455-4bd9-a296-6c772478486a)

### Reproducer

https://stackblitz.com/github/ivanpajon/bug-fileupload-component?file=app%2Fcomponent.tsx

### PrimeReact version

10.2.1

### React version

18.x

### Language

TypeScript

### Build / Runtime

Next.js

### Browser(s)

ALL

### Steps to reproduce the behavior

1. Open Stackblitz bug reproducer
2. Attach same image to 4 FileUpload components
3. Click submit button
4. See in Stackblitz console (not browser, because it is a Server Action) printed logs
5. Check that first FileUpload sent null, second and third sent wrong image, and fourth sent correct image only with hidden input

### Expected behavior

If I set an Id and a Name to FileUpload component, it should set it to its internal hidden input.
And when I choose a file and submit the form, the file should be set correctly.

**Comment by melloware** [comment]
@ivanpajon interesting I have never had an issue with FileUpload but I have always followed the examples: https://primereact.org/fileupload/#advanced

It looks like Advanced mode wants to upload to a URL and you have no URL you are trying to use it with Form submission?

See: https://primereact.org/fileupload/#api.FileUpload.props.url

So I think you were trying to use it for a use case it was not intended.

**Comment by ivanpajon** [comment]
Yes, you are right. I am trying to use as a normal file selector inside a form.

I spent time this days ago trying to get a local fork of primereact working so maybe I can make some changes and create a pull request but it was impossible for me... I followed instructions in DEVELOPMENT.md, link and build correctly (via npm run dev:link:windows as I am using Windows), but when I serve my project I get this error. Switch from local dependency to npm package make it works with no changes in my project.
![image](https://github.com/primefaces/primereact/assets/17519714/0001dd6d-2d40-4e43-ba6c-5c6fce17f157)
